kvdb/sqlbase/sqlerrors_sqlite.go

```diff
@@ -13,6 +14,17 @@ import (
 // parseSqliteError attempts to parse a sqlite error as a database agnostic
 // SQL error.
 func parseSqliteError(err error) error {
+	// If the error isn't wrapped properly, the errors.As call with fail,
+	// so we'll also try to check the expected error message directly.
+	// This is taken from:
+	// https://gitlab.com/cznic/sqlite/-/blob/v1.25.0/sqlite.go#L75.
+	const sqliteErrMsg = "SQLITE_BUSY"
+	if strings.Contains(err.Error(), sqliteErrMsg) {
```

Comment:
Do we need a `if err == nil {` safety check first or are we guaranteed to only hit this on a non-nil error?